cmd/gitops/install/cmd.go

```diff
@@ -74,97 +76,64 @@ func installRunCmd(cmd *cobra.Command, args []string) error {
-		return errors.New("Weave GitOps does not yet support installation onto a cluster that is using Flux.\nPlease uninstall flux before proceeding:\n  $ flux uninstall")
-	case kube.Unknown:
-		return fmt.Errorf("Weave GitOps cannot talk to the cluster %s", clusterName)
-	}
-
-	clusterApplier := applier.NewClusterApplier(k)
-
-	var gitClient git.Git
-
-	var gitProvider gitproviders.GitProvider
-
-	factory := services.NewFactory(flux, log)
-
-	if err != nil {
-		return fmt.Errorf("failed getting kube service: %w", err)
-	}
-
 	if installParams.DryRun {
-		gitProvider, err = gitproviders.NewDryRun()
+		manifests, err := models.BootstrapManifests(fluxClient, clusterName, namespace, configURL)
 		if err != nil {
-			return fmt.Errorf("error creating git provider for dry run: %w", err)
+			return fmt.Errorf("failed getting gitops manifests: %w", err)
 		}
-	} else {
-		_, err = flux.Install(namespace, false)
-		if err != nil {
-			return err
-		}
-
-		providerClient := internal.NewGitProviderClient(osysClient.Stdout(), osysClient.LookupEnv, auth.NewAuthCLIHandler, log)
 
-		gitClient, gitProvider, err = factory.GetGitClients(context.Background(), providerClient, services.GitConfigParams{
-			URL:       installParams.ConfigRepo,
-			Namespace: namespace,
-			DryRun:    installParams.DryRun,
-		})
-
-		if err != nil {
-			return fmt.Errorf("error creating git clients: %w", err)
+		for _, manifest := range manifests {
+			fmt.Println(string(manifest.Content))
 		}
-	}
-
-	cluster := models.Cluster{Name: clusterName}
-	repoWriter := gitrepo.NewRepoWriter(configURL, gitProvider, gitClient, log)
-	automationGen := automation.NewAutomationGenerator(gitProvider, flux, log)
-	gitOpsDirWriter := gitopswriter.NewGitOpsDirectoryWriter(automationGen, repoWriter, osysClient, log)
 
-	clusterAutomation, err := automationGen.GenerateClusterAutomation(ctx, cluster, configURL, namespace)
-	if err != nil {
-		return err
+		return nil
 	}
 
-	wegoConfigManifest, err := clusterAutomation.GenerateWegoConfigManifest(clusterName, namespace, namespace)
-	if err != nil {
-		return fmt.Errorf("failed generating wego config manifest: %w", err)
-	}
+	log := internal.NewCLILogger(os.Stdout)
+	providerClient := internal.NewGitProviderClient(osysClient.Stdout(), osysClient.LookupEnv, auth.NewAuthCLIHandler, log)
 
-	manifests := append(clusterAutomation.Manifests(), wegoConfigManifest)
+	factory := services.NewFactory(fluxClient, log)
 
-	if installParams.DryRun {
-		for _, manifest := range manifests {
-			log.Println(string(manifest.Content))
-		}
+	// We temporarily need this here otherwise GetGitClients is going to fail
+	// as it needs the namespace created to apply the secret
+	namespaceObj := &corev1.Namespace{}
+	namespaceObj.Name = namespace
 
-		return nil
+	if err := kubeClient.Raw().Create(ctx, namespaceObj); err != nil {
+		if !strings.Contains(err.Error(), "already exists") {
+			return fmt.Errorf("failed creating namespace %s: %w", namespace, err)
```

Comment:
@palemtnrider  The idea is to fail here in case an error occurs but only if that error is different from the "already exists" error. Also if this fails the very first time and the namespace doesn't exist, we need to know what happened. What kind of recoverable errors do you have in mind?. Maybe we can just add those cases. Or we could also first check if the namespace exists already to decide to call `Create` or not? That way we skip attempting to create the cluster each consequent time we run install.

Reply:
Thanks for the clarification.